File: src/main/java/com/ukm/programkerja/service/TempatService.java
package com.ukm.programkerja.service;

import com.ukm.programkerja.dto.request.TempatRequest;
import com.ukm.programkerja.dto.response.TempatResponse;

import java.util.List;

public interface TempatService {

    TempatResponse create(TempatRequest request);

    TempatResponse update(Long id, TempatRequest request);

    void delete(Long id);

    TempatResponse getById(Long id);

    List<TempatResponse> getAll();
}

File: frontend/src/services/tempatService.js
import api from './api'

export default {
  getAll() {
    return api.get('/tempat')
  },
  getById(id) {
    return api.get(`/tempat/${id}`)
  },
  create(data) {
    return api.post('/tempat', data)
  },
  update(id, data) {
    return api.put(`/tempat/${id}`, data)
  },
  delete(id) {
    return api.delete(`/tempat/${id}`)
  },
}
